followthemoney/graph.py

```diff
@@ -129,39 +150,41 @@ class Graph(object):
     backends, like Aleph.
     """
 
-    def __init__(self, edge_types=registry.pivots):
-        edge_types = registry.get_types(edge_types)
-        self.edge_types = [t for t in edge_types if t.matchable]
+    def __init__(self, edge_types: Sequence[Union[str, PropertyType]]=registry.pivots):
+        _edge_types = registry.get_types(edge_types)
+        self.edge_types: List[PropertyType] = [t for t in _edge_types if t.matchable]
         self.flush()
 
     def flush(self):
-        self.edges = {}
-        self.nodes = {}
-        self.proxies = {}
+        self.edges: Dict[str, Edge] = {}
+        self.nodes: Dict[str, Node] = {}
+        self.proxies: Dict[str, Optional[EntityProxy]] = {}
 
-    def queue(self, id_, proxy=None):
+    def queue(self, id_: str, proxy: Optional[EntityProxy]=None):
         if id_ not in self.proxies or proxy is not None:
             self.proxies[id_] = proxy
 
     @property
-    def queued(self):
+    def queued(self) -> List[str]:
         return [i for (i, p) in self.proxies.items() if p is None]
 
-    def _get_node_stub(self, prop, value):
+    def _get_node_stub(self, prop: Property, value: str) -> Node:
         if prop.type == registry.entity:
             self.queue(value)
         node = Node(prop.type, value, schema=prop.range)
         if node.id not in self.nodes:
             self.nodes[node.id] = node
         return self.nodes[node.id]
 
-    def _add_edge(self, proxy, source, target):
-        source = self._get_node_stub(proxy.schema.source_prop, source)
-        target = self._get_node_stub(proxy.schema.target_prop, target)
-        edge = Edge(self, source, target, proxy=proxy)
+    def _add_edge(self, proxy: EntityProxy, source: str, target: str):
+        if proxy.schema.source_prop is None or proxy.schema.target_prop is None:
```

Comment:
I don't think either of these checks are needed given how it's invoked.

Reply:
I can add a type ignore on both calls to `self._get_node_stub`, but as `proxy.schema.source_prop` and `proxy.schema.target_prop` can be `None`, mypy will not like it if I don't do the test first.